x/stakeibc/keeper/reward_converter_test.go

```diff
@@ -33,6 +33,184 @@ type TransferRewardHostToTradeTestCase struct {
+			expectedStrideFeeAmount: sdkmath.NewInt(90),
+			expectedReinvestAmount:  sdkmath.NewInt(900),
+		},
+		{
+			// (Example #1 but with a 2x larger TVL)
+			// 10 CP Liquid Stake, 200 TVL => 5% contribution
+			// 1000 rewards, 10% stride fee => 100 total fees
+			// 100 total fees * 5% contribution * 50% rebate
+			// => 2.5 rebate (truncated to 2), 98 stride fee, 998 reinvest
+			name:                     "case 3",
+			communityPoolLiquidStake: sdk.NewInt(10),
+			totalDelegations:         sdkmath.NewInt(200),
+			rewardAmount:             sdkmath.NewInt(1000),
+			strideFee:                10,
+			rebatePercentage:         sdk.MustNewDecFromStr("0.5"),
+
+			expectedRebateAmount:    sdkmath.NewInt(2),
+			expectedStrideFeeAmount: sdkmath.NewInt(98),
+			expectedReinvestAmount:  sdkmath.NewInt(900),
+		},
+		{
+			// (Example #1 but with a 2x larger fee)
+			// 10 CP Liquid Stake, 100 TVL => 10% contribution
+			// 1000 rewards, 20% stride fee => 200 total fees
+			// 200 total fees * 10% contribution * 50% rebate
+			// => 10 rebate, 190 stride fee, 800 reinvest
+			name:                     "case 4",
+			communityPoolLiquidStake: sdk.NewInt(10),
+			totalDelegations:         sdkmath.NewInt(100),
+			rewardAmount:             sdkmath.NewInt(1000),
+			strideFee:                20,
+			rebatePercentage:         sdk.MustNewDecFromStr("0.5"),
+
+			expectedRebateAmount:    sdkmath.NewInt(10),
+			expectedStrideFeeAmount: sdkmath.NewInt(190),
+			expectedReinvestAmount:  sdkmath.NewInt(800),
+		},
+		{
+			// (Example #1 but with a smaller rebate)
+			// 10 CP Liquid Stake, 100 TVL => 10% contribution
+			// 1000 rewards, 10% stride fee => 100 total fees
+			// 100 total fees * 10% contribution * 20% rebate
+			// => 2 rebate, 98 stride fee, 998 reinvest
+			name:                     "case 5",
+			communityPoolLiquidStake: sdk.NewInt(10),
+			totalDelegations:         sdkmath.NewInt(100),
+			rewardAmount:             sdkmath.NewInt(1000),
+			strideFee:                10,
+			rebatePercentage:         sdk.MustNewDecFromStr("0.2"),
+
+			expectedRebateAmount:    sdkmath.NewInt(2),
+			expectedStrideFeeAmount: sdkmath.NewInt(98),
+			expectedReinvestAmount:  sdkmath.NewInt(900),
+		},
+		{
+			// (Example #1 but with a larger rebate)
+			// 10 CP Liquid Stake, 100 TVL => 10% contribution
+			// 1000 rewards, 10% stride fee => 100 total fees
+			// 100 total fees * 10% contribution * 79% rebate
+			// => 7.9 rebate (truncated to 7), 93 stride fee, 993 reinvest
+			name:                     "case 6",
+			communityPoolLiquidStake: sdk.NewInt(10),
+			totalDelegations:         sdkmath.NewInt(100),
+			rewardAmount:             sdkmath.NewInt(1000),
+			strideFee:                10,
+			rebatePercentage:         sdk.MustNewDecFromStr("0.79"),
+
+			expectedRebateAmount:    sdkmath.NewInt(7),
+			expectedStrideFeeAmount: sdkmath.NewInt(93),
+			expectedReinvestAmount:  sdkmath.NewInt(900),
+		},
+		{
+			// No rebate - all should be reinvested
```

Comment:
nit: add case where entire TVL is comm pool TVL

Reply:
added